### PR TITLE
i.ifft: Add Test Suite

### PR DESCRIPTION
**Pull request by jayneel-shah18**
This PR introduces a test suite for the `i.ifft` GRASS GIS module. The test suite is aimed at verifying the module’s core functionality, confirming statistical accuracy, and testing various edge cases to ensure stable and reliable performance.

### **Key Updates in this PR:**
- Validate IFFT output for linearity, scaling, coefficient symmetry, masking, and reconstruction.
- Ensure correct handling of input rasters for real and imaginary components.
- Test IFFT behavior under different masking and raster configurations.

### **Test Case Additions**
- **Linearity Property**: Test the linearity of the inverse fast Fourier transform (IFFT) by combining two rasters (test_linearity_property).
- **Scaling Property**: Validate the scaling property of the IFFT to ensure `ifft(2.5 * x) == 2.5 * ifft(x)` (test_scaling_property).
- **Coefficient Symmetry**: Test if the IFFT preserves symmetry in the real-valued output (test_coefficient_symmetry).
- **Mask Functionality**: Ensure proper behavior when masking is applied (test_mask_functionality).
- **IFFT Reconstruction**: Check if the IFFT can accurately reconstruct the original raster from the FFT (test_ifft_reconstruction).

### **Performance**
The test suite execution times:
- **Windows**: 3.137s
- **macOS**: 2.027s
- **Ubuntu**: 2.555s

**This test suite enhances the reliability of the `i.ifft` module by covering a broad spectrum of cases. Looking forward to feedback and potential refinements!**

### CODE REVIEW

File: imagery/i.ifft/testsuite/test_i_ifft.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        )
+        self.assertRasterExists(self.output_raster)
+
+        self.assertModule(
+            "i.ifft",
+            real="real_input2",
+            imaginary="imag_input2",
+            output="output_raster2",
+            overwrite=True,
+        )
```
Comment (petrasovaa) [inline imagery/i.ifft/testsuite/test_i_ifft.py:67]: What are these 2 i.ifft runs testing?

File: imagery/i.ifft/testsuite/test_i_ifft.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+            real="real_sym",
+            imaginary="imag_zero",
+            output="output_sym",
+            overwrite=True,
+        )
+        self.assertRasterExists("output_sym")
+        self.temp_rasters.append("output_sym")
+
+        self.runModule(
+            "r.mapcalc", expression="shifted = output_sym[5, 5]", overwrite=True
```
Comment (petrasovaa) [inline imagery/i.ifft/testsuite/test_i_ifft.py:142]: Could you explain this test more?

File: imagery/i.ifft/testsuite/test_i_ifft.py (hunk 1 above)
Comment (jayneel-shah18) [reply imagery/i.ifft/testsuite/test_i_ifft.py:67]: These two `i.ifft` runs generate two separate rasters by performing the inverse Fourier transform on different input sets. The outputs are then combined into a third raster (`output_raster_sum`) using the summed inputs (`real_input_sum`, `imag_input_sum`). Finally, the results of `output_raster_sum` are compared against reference statistics, which were derived using the same method. This test validates the linearity property of the inverse Fourier transform by ensuring that `IFFT(A) + IFFT(B) = IFFT(A + B)`.

File: imagery/i.ifft/testsuite/test_i_ifft.py (hunk 2 above)
Comment (jayneel-shah18) [reply imagery/i.ifft/testsuite/test_i_ifft.py:142]: This test checks whether `i.ifft` preserves symmetry in the real-valued output. The input (`real_sym`) is designed to be symmetric around column 5, and `imag_zero` ensures there is no imaginary component. I wanted to compare the flipped raster with the raster generated by `i.ifft` and initially, I attempted to use `r.flip` to create the shifted raster, but it did not work as expected. So instead, I used `shifted = output_sym[5, 5]` to obtain a reference value and computed `sym_diff = abs(output_sym - shifted)` to find deviations from symmetry. Then this is compared against reference statistics which validates that the IFFT output maintains the expected symmetry.

File: imagery/i.ifft/testsuite/test_i_ifft.py (hunk 1 above)
Comment (petrasovaa) [reply imagery/i.ifft/testsuite/test_i_ifft.py:67]: That's nice but that's not what the code is doing...

Please include a short version of this explanation in the test function documentation

File: imagery/i.ifft/testsuite/test_i_ifft.py (hunk 2 above)
Comment (jayneel-shah18) [reply imagery/i.ifft/testsuite/test_i_ifft.py:133]: I have included a short version of this explanation in the test documentation.

Comment (petrasovaa) [reply imagery/i.ifft/testsuite/test_i_ifft.py:133]: Could you explain how the results validate the symmetry, I don't see it. Note that shifted raster has all zeros for me in the upper left quarter of the raster and rest is null.

File: imagery/i.ifft/testsuite/test_i_ifft.py (hunk 1 above)
Comment (petrasovaa) [reply imagery/i.ifft/testsuite/test_i_ifft.py]: OK, I realize what you wanted to do. To test `IFFT(A) + IFFT(B) = IFFT(A + B)` simply run the 3 i.fft, add the 2 maps and compare the maps directly with assertRastersEqual.

File: imagery/i.ifft/testsuite/test_i_ifft.py (hunk 2 above)
Comment (jayneel-shah18) [reply imagery/i.ifft/testsuite/test_i_ifft.py:133]: Thank you for your feedback. I see that defining shifted as output_sym[5,5] does not fully represent the intended symmetry check, it only confirms local consistency around (5,5), not global symmetry across the entire raster. I was able to do the intended check using `r.flip` where I flip the raster horizontally and compare the output and flipped raster. I have updated the test now. Thank you!

Comment (petrasovaa) [reply imagery/i.ifft/testsuite/test_i_ifft.py]: r.flip is an addon, so it won't run in the CI, maybe you can use numpy?

File: imagery/i.ifft/testsuite/test_i_ifft.py (hunk 1 above)
Comment (jayneel-shah18) [reply imagery/i.ifft/testsuite/test_i_ifft.py:71]: I have updated the test to perform 3 i.ifft runs and directly compare the resulting maps.

File: imagery/i.ifft/testsuite/test_i_ifft.py (hunk 2 above)
Comment (jayneel-shah18) [reply imagery/i.ifft/testsuite/test_i_ifft.py]: Made the change to use `np.flip` instead of `r.flip`.

File: imagery/i.ifft/testsuite/test_i_ifft.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+        self.runModule(
+            "r.mapcalc", expression="diff = output_sym - mirrored", overwrite=True
+        )
+        self.temp_rasters.append("diff")
+        reference_stats = {
+            "min": -16.527864,
+            "max": 16.527864,
+            "mean": 0,
+            "sum": 0,
+        }
```
Comment (petrasovaa) [inline imagery/i.ifft/testsuite/test_i_ifft.py:170]: This should be zero, no? This may point to some problem in the tool.

Comment (jayneel-shah18) [reply imagery/i.ifft/testsuite/test_i_ifft.py:170]: Yes, all the stats should ideally be zero. I even tested with slightly larger rasters, and while the stats for the output and mirrored rasters are equal, the difference raster still shows nonzero values for min and max. This might indicate an issue with the tool.

Comment (petrasovaa) [reply imagery/i.ifft/testsuite/test_i_ifft.py:170]: So then you need to document it in the test, that this test shows potential problem.

**Review by petrasovaa — APPROVED** [review]
